File: 209_minimum_size_subarray_sum/minSubArrayLen.py
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)

        if n == 0:
            return 0

        i, j, s, res = 0, 0, 0, n + 1

        while j < n:
            s += nums[j]
            j += 1

            while s >= target:
                res = min(res, j - i)
                s -= nums[i]
                i += 1

        return res if res < n + 1 else 0

File: 209_minimum_size_subarray_sum/minSubArrayLen.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int j = 0;
        int i = 0;
        int sum = 0;
        int res = n + 1;

        while (j < n) {
            sum += nums[j];
            j++;

            while (sum >= target) {
                res = Math.min(res, j - i);
                sum -= nums[i];
                i++;
            }
        }

        return res <= n ? res : 0;
    }
}